sustainml_cpp/include/sustainml_cpp/orchestrator/OrchestratorNode.hpp

```diff
@@ -109,6 +109,11 @@ class OrchestratorNode
             const NodeID& node_id,
             void*& data);
 
+    RetCode_t get_task_data_(
+            const types::TaskId& task_id,
+            const NodeID& node_id,
+            void* data);
```

Comment:
LOL, it is weird that using this new method the whole thing works fine. In normal C++, `data` would never be populated but it seems that python is directly passing a reference to the `data`'s pointer (as the method above does) instead of doing a copy by value. We need to ensure that using the method above does not work, because it should be the reasonable one to use, or at least, change it to `void**`. 
If we finally need to keep this method, we need to appropriately doxygen it .

Reply:
Yes, it seems Python does not support references, but assignments. I didn't investigate more in-depth alternatives, and I don't like the approach I followed but at least wanted to retrieve the data somehow.